presto-main/src/main/java/com/facebook/presto/server/security/OpenLdapBinder.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.server.security;
+
+import javax.inject.Inject;
+
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class OpenLdapBinder
+        implements LdapBinder
+{
+    private final String baseDistinguishedName;
+
+    @Inject
+    public OpenLdapBinder(LdapServerConfig config)
+    {
+        baseDistinguishedName = requireNonNull(config.getBaseDistinguishedName(), "baseDistinguishedName is null");
+    }
+
+    @Override
+    public String getBindDistinguishedName(String user)
+    {
+        return format("uid=%s,%s", user, baseDistinguishedName);
```

Comment:
I think we can get rid of the separate binders and simply have the config be a format string of some sort. So for OpenLDAP, you would specify something like `uid=${USER},dc=example,dc=net`